api/specs/webserver/openapi-projects-comments.yaml

```diff
@@ -0,0 +1,265 @@
+paths:
+  /projects/{project_uuid}/comments:
+    get:
+      tags:
+      - project
+      - comments
+      summary: Retrieve all comments for a specific project.
+      operationId: list_project_comments
+      parameters:
+      - required: true
+        schema:
+          title: Project Uuid
+          type: string
+          format: uuid
+        name: project_uuid
+        in: path
+      - required: false
+        schema:
+          title: Limit
+          minimum: 0
+          type: integer
+          default: 20
+        name: limit
+        in: query
+      - required: false
+        schema:
+          title: Offset
+          minimum: 0
+          type: integer
+          default: 0
+        name: offset
+        in: query
+      responses:
+        '200':
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Envelope_list_models_library.projects_comments.ProjectsCommentsAPI__'
+    post:
+      tags:
+      - project
+      - comments
+      summary: Create a new comment for a specific project. The request body should
+        contain the comment content and user information.
+      operationId: create_project_comment
+      parameters:
+      - required: true
+        schema:
+          title: Project Uuid
+          type: string
+          format: uuid
+        name: project_uuid
+        in: path
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/_CreateProjectCommentsBodyParams'
+        required: true
+      responses:
+        '201':
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Envelope_PositiveInt_'
+  /projects/{project_uuid}/comments/{comment_id}:
+    get:
+      tags:
+      - project
+      - comments
+      summary: Retrieve a specific comment by its ID within a project.
+      operationId: get_project_comment
+      parameters:
+      - required: true
+        schema:
+          title: Project Uuid
+          type: string
+          format: uuid
+        name: project_uuid
+        in: path
+      - required: true
+        schema:
+          title: Comment Id
+          exclusiveMinimum: true
+          type: integer
+          minimum: 0
+        name: comment_id
+        in: path
+      responses:
+        '200':
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Envelope_ProjectsCommentsAPI_'
+    put:
+      tags:
+      - project
+      - comments
+      summary: Update the content of a specific comment for a project. The request
```

Comment:
Question: can you change the comments of others? what is the concept here? same as in Github?

Reply:
I guess in the first iteration, Odei will not implement the updating of the comments on the frontend. But I have a feeling that the request will come so already prepared all basic CRUD operations which should be able to handle also updating. Yes currently updating and deleting can be done by anybody who has access to the study (similarly as in GitHub) but this behavior cab be changed easily if requested.